src/firewall/core/fw.py

```diff
@@ -1069,12 +1069,7 @@ def check_policy(self, policy):
         return _policy
 
     def check_zone(self, zone):
-        _zone = zone
-        if not _zone or _zone == "":
-            _zone = self.get_default_zone()
-        if _zone not in self.zone.get_zones():
-            raise FirewallError(errors.INVALID_ZONE, _zone)
-        return _zone
+        return self.zone.check_zone(zone)
```

Comment:
Can we remove this function entirely? Callers can do `fw.zone.check_zone()` directly.

Reply:
yes.

I left it for now, because I didn't want the branch to become even larger.

What I care here, that `fw.py`'s check function has essentially no more logic. What makes a valid zone, is now entirely delegated to `self.zone.check_zone()`.